test/jdk/javax/imageio/plugins/jpeg/WriteJPEGThumbnailTest.java

```diff
@@ -0,0 +1,160 @@
+/*
+ * Copyright (c) 2025, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/**
+ * @test
+ * @bug     8351110
+ * @summary Test verifies that when a JFIF thumbnail may exceed 65535 bytes
+ *          we still write a valid JPEG file by clipping the thumbnail.
+ * @run     main WriteJPEGThumbnailTest
+ */
+
+import javax.imageio.IIOImage;
+import javax.imageio.ImageIO;
+import javax.imageio.ImageReader;
+import javax.imageio.ImageWriter;
+import javax.imageio.stream.ImageInputStream;
+import javax.imageio.stream.ImageOutputStream;
+import java.awt.geom.AffineTransform;
+import java.awt.image.BufferedImage;
+import java.awt.Color;
+import java.awt.Graphics2D;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Iterator;
+import java.util.List;
+
+public class WriteJPEGThumbnailTest {
+
+    private static void assertEquals(int expected, int observed) {
+        if (expected != observed) {
+            throw new Error("expected " + expected + ", but observed " + observed);
+        }
+    }
+
+    public static void main(String[] args) throws Exception {
+        // this always passed. 21800 * 3 = 65400, which fits in a 65535-byte segment.
+        boolean b1 = new WriteJPEGThumbnailTest(100, 218).run();
+
+        // this failed prior to resolving 8351110. 21900 * 3 = 65700, which is too large
+        // for a JPEG segment. Now we clip the thumbnail to make it fit. (Previously
+        // we wrote a corrupt JPEG file.)
+        boolean b2 = new WriteJPEGThumbnailTest(100, 219).run();
+        if (!(b1 && b2)) {
+            System.err.println("Test failed");
+            throw new Error("Test failed");
+        }
+    }
+
+    final int thumbWidth, thumbHeight;
```

Comment:
```suggestion
    final int thumbWidth;
    final int thumbHeight;
```
I prefer them on separate lines, although it's not that important for a test.